autoextract_poet/page_inputs.py

```diff
@@ -38,34 +38,44 @@ class _AutoExtractData(Generic[T]):
 
     https://doc.scrapinghub.com/autoextract.html#responses
     """
-
-    item_key: ClassVar[str]
-
+    page_type: ClassVar[str]
     data: dict
+    _item: Optional[T] = attr.ib(init=False, repr=False)
+
+    def __attrs_post_init__(self):
+        # Caching item to avoid building it again and again
+        self._item = self.item_class.from_dict(self.data[self.page_type])
 
     @property
     def item_class(self):
         return self.__orig_bases__[0].__args__[0]
 
     def to_item(self) -> Optional[T]:
-        return self.item_class.from_dict(self.data[self.item_key])
+        return self._item
```

Comment:
this makes the returned item mutable - page.to_item() may return a different result second time, which doesn't represent the exracted item. What's the use case for having this cache by default - when would you need to call to_item() multiple times, and reusing the result of the first call is not an option?

Reply:
This might be a premature optimization by my side. The reasons to follow this direction from my side are the following:
* Nothing prevents the user pass the `AutoExtract<X>Data` instance around the code and to call `to_item` several times. Maybe this is not going to be common in practice, so it could be an overoptimization by my side. 
* If fails sooner, as it builds the item in the constructor. 

But yes, maybe is better to keep it simple, and create the object always in the `to_item`. Let me see if I have the time to introduce the changes today.